core/src/message/init.rs

```diff
@@ -115,8 +115,6 @@ impl InitMessage {
 /// This structure is preparation for future init message sending. Has no message id.
 #[derive(Clone, Default, Debug, Eq, Hash, Ord, PartialEq, PartialOrd, Decode, Encode, TypeInfo)]
 pub struct InitPacket {
-    /// Newly created program id.
-    program_id: ProgramId,
```

Comment:
why did you remove the field?

Reply:
programid needs messageid, so i cannot generate it until a messageid is also generated, this was the whole thing I was doing in this PR

Reply:
you need to use message id of the message during which processing create_program was called